### ISSUE TITLE
Model size difference when exporting from pytorch to onnx

### ISSUE THREAD

**Issue by YOOHYOJEONG** [ultralytics/ultralytics#17279]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,
I did benchmark COCO through the command below using pre-trained model.   
```
yolo benchmark model=/ultralytics/DATA/weights/coco_pretrained/yolov11/yolov11n.pt data=/ultralytics/DATA/weights/coco_pretrained/coco.yaml    
```
   
The result is as shown in the image below.   
<img width="735" alt="image" src="https://github.com/user-attachments/assets/56350cdb-314a-4d9c-aa6d-38cba3c2eb3c">

I have a question after seeing this result.   
ONNX models are generally smaller than PyTorch models because they omit unnecessary components (like training details) and focus solely on the inference graph. Right?   
But, The ultralytics benchmark results are not.    

`half=True`  I can only use this option to be similar in size and not smaller. Even the speed of inference has become slower for ONNX.   
![image](https://github.com/user-attachments/assets/b6a6fc6b-bf55-4b8b-9fe6-a67517591392)   

Can you review if there is a problem with the export code?   
   

   
    
`yolo checks
`Ultralytics 8.3.18 🚀 Python-3.11.9 torch-2.4.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4090, 24217MiB)
Setup complete ✅ (32 CPUs, 125.5 GB RAM, 1549.8/1832.2 GB disk)

OS Linux-6.8.0-45-generic-x86_64-with-glibc2.35
Environment Linux
Python 3.11.9
Install git
RAM 125.51 GB
Disk 1549.8/1832.2 GB
CPU Intel Core(TM) i9-14900K
CPU count 32
GPU NVIDIA GeForce RTX 4090, 24217MiB
GPU count 2
CUDA 12.1

numpy ✅ 1.23.5>=1.23.0
matplotlib ✅ 3.9.2>=3.3.0
opencv-python ✅ 4.10.0.84>=4.6.0
pillow ✅ 10.2.0>=7.1.2
pyyaml ✅ 6.0.2>=5.3.1
requests ✅ 2.32.3>=2.23.0
scipy ✅ 1.14.1>=1.4.1
torch ✅ 2.4.1+cu121>=1.8.0
torchvision ✅ 0.19.1+cu121>=0.9.0
tqdm ✅ 4.66.5>=4.64.0
psutil ✅ 6.0.0
py-cpuinfo ✅ 9.0.0
pandas ✅ 2.2.3>=1.1.4
seaborn ✅ 0.13.2>=0.11.0
ultralytics-thop ✅ 2.0.9>=2.0.0
torch ✅ 2.4.1+cu121!=2.4.0,>=1.8.0; sys_platform == "win32"

**Comment by UltralyticsAssistant** [comment]
👋 Hello @YOOHYOJEONG, thank you for bringing this to our attention and for your interest in Ultralytics 🚀! We're excited to help you resolve your query.

For new users or anyone looking for more information, we highly recommend checking out the [Docs](https://docs.ultralytics.com) where you'll find numerous examples for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many common questions are addressed there as well.

Regarding your question about the model size difference between PyTorch and ONNX exports, ONNX models usually focus solely on the inference graph, which can lead to size and performance variations. However, inconsistency or unexpected results could be a bug 🐛. To help us investigate further, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) if you haven't already.

### Upgrade

To ensure you're using the latest features and bug fixes, please upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

### Environments

For optimal experience, you can run YOLO in verified environments with pre-installed dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Community & Support

Join our vibrant community for additional support and discussion. Real-time chat is available on [Discord](https://ultralytics.com/discord) 🎧, or you can engage in deeper discussions on [Discourse](https://community.ultralytics.com) or share insights on our [Subreddit](https://reddit.com/r/ultralytics). 

### Status

We ensure our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are running correctly for various [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/), across different platforms.

Rest assured, this is an automated response to guide you initially, and a dedicated Ultralytics engineer will also assist you shortly. Thank you for your patience and collaboration in enhancing the Ultralytics experience! 😊

**Comment by Y-T-G** [comment]
The PyTorch weights are in FP16 format. So exporting without `half=True` would make it almost twice as large because it's using FP32.

**Comment by YOOHYOJEONG** [comment]
When exporting to `half=True`, there is not much difference in size, is this normal?
I think the size needs to be reduced a lot when converting to onnx.

**Comment by Y-T-G** [comment]
There wouldn't be much difference because the original `.ptt` weights is FP16. So you're going from FP16 to FP16.

**Comment by YOOHYOJEONG** [comment]
Oh.. I got it. 
Thank you !

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.
